Src/FluentAssertions/Collections/GenericCollectionAssertions.cs

```diff
@@ -317,17 +317,20 @@ public AndConstraint<TAssertions> BeEmpty(string because = "", params object[] b
 
             EquivalencyAssertionOptions<IEnumerable<TExpectation>> options = config(AssertionOptions.CloneDefaults<TExpectation>()).AsCollection();
 
-            var context = new EquivalencyValidationContext(Node.From<IEnumerable<TExpectation>>(() => CallerIdentifier.DetermineCallerIdentity()))
+            var context = new EquivalencyValidationContext(Node.From<IEnumerable<TExpectation>>(CallerIdentifier.DetermineCallerIdentity), options)
```

Comment:
What makes you say that?

Reply:
As in whether it's true or why I point it out?

Here's an example of the difference in the generated code.
https://sharplab.io/#v2:EYLgtghglgdgPgAQEwEYCwAoTCDMACBFANgKTwGEIAbKgUwCcBJAE1pgBcoAzKBzAb0x5hBfIRLAA9pKp4AIrXYMwsWpRoMWbTuwCeACgCUeALwA+PO3oBXWgG5MAX0zYxZcgKEiADvSgA3CCUCYjwpGTwAMWl9BABWAB5wqgtWKmNzPDSjBwwRPC9hXDDpWQA5SRhKAGMAC1pmI1MLaMl9dTomVg5uXnoAOgUlehUYNWpOrR69Q1z8wtESiJr6xoyWmKbMjs1uzh4GQcVlVR2u7Sg9I1mnIA===

I point it out, because I think we should benefit from the compiler caching.

Reply:
You seem to be right, although I ran the benchmarks a couple of times between my branch and `develop`, and there's no noticeable difference. What's also weird that Rider/R# are both suggesting to use a method group.

Reply:
I fully agree with any person/analyzer that finds method groups prettier, but the improvements does not seem to be come in the near future https://github.com/dotnet/roslyn/issues/5835.

What I think about, every time I see a method wrapped in a lambda expression.
![image](https://user-images.githubusercontent.com/919634/118410030-bea41380-b68d-11eb-9d74-6c8bc21662f8.png)

Reply:
Although I'm not convinced that this performance improvement is noticeable, I'm willing to revert my changes for your sake. But then we should also disable the suggestion from Rider/R#.

Reply:
Feel free to disable `IDE0005` as well.
Then I'll (probably) stop complaining about unused `using`s.

Reply:
What does `IDE0005` have to do with this PR?

Reply:
nothing, just ignore me here.